index.js

```diff
@@ -1,46 +1,24 @@
 'use strict';
-var execFile = require('child_process').execFile;
-var eachAsync = require('each-async');
-var appPath = require('app-path');
-var bundleId = require('bundle-id');
-var osxAppVersion = require('osx-app-version');
-var appSize = require('app-size');
+const appPath = require('app-path');
+const bundleId = require('bundle-id');
+const osxAppVersion = require('osx-app-version');
+const appSize = require('app-size');
 
-module.exports = function (app, cb) {
+module.exports = app => {
 	if (process.platform !== 'darwin') {
-		throw new Error('Only OS X systems are supported');
+		return Promise.reject(new Error('Only OS X is supported'));
 	}
 
 	if (typeof app !== 'string') {
-		throw new Error('Application is required');
+		return Promise.reject(new Error('Application is required'));
 	}
 
-	var obj = {};
-
-	eachAsync([osxAppVersion, bundleId, appPath, appSize], function (fn, index, done) {
-		fn(app, function(err, res) {
-			if (err) {
-				done(err);
-			}
-
-			if (index === 0) {
-				obj.version = res;
-			} else if (index === 1) {
-				obj.bundle = res;
-			} else if (index === 2) {
-				obj.path = res;
-			} else {
-				obj.size = res;
-			}
-
-			done();
-		});
-	}, function (err) {
-		if (err) {
-			cb(err);
-			return;
-		}
-
-		cb(null, obj);
+	return Promise.all([osxAppVersion(app), bundleId(app), appPath(app), appSize(app)]).then(res => {
+		return {
+			version: res[0],
+			bundle: res[1],
+			path: res[2],
+			size: res[3]
```

Comment:
Hmm, I rather use the following:

1. `path`
2. `version`
3. `bundle`
4. `size`